### ISSUE TITLE
Using STIR as an alternative to FRI

### ISSUE THREAD

**Issue by Antonio95** [Plonky3/Plonky3#485]
[STIR](https://eprint.iacr.org/2024/390) is a low-degree test for univariate polynomials, much like FRI. A PR was recently opened in this repo to include it (in its PCS form) in the README: #351 . It might make sense to fully add it to the toolbox in parallel to FRI, as a plug-and-play replacement for it (perhaps even under some kind of trait umbrella to make it easier to choose between one and the other).

Compared to FRI, STIR has shorter proofs because of its lower number of queries (`O(lambda * log(log(d)))` for polys of degree < d, as opposed to `O(lambda *  log(d))`) and involves less verifier-side hashing (thus being more recursion-friendly) but incurs a slight increase in prover time.

In terms of implementation work, STIR is rather similar to FRI but uses quotients for out-of-domain sampling as well as slightly different folding. It's possible most of the necessary pieces are already present in the repo in one way or another. Of course, aside from native execution, it would be interesting to add a recursive verifier, which might be a bit more involved.

Is STIR already being worked on/something that there is interest in?

**Comment by Nashtare** [comment]
Closing as superseded by #1647
